src/main/java/com/google/pubsublite/kafka/sink/PublisherFactoryImpl.java

```diff
@@ -15,25 +15,42 @@
  */
 package com.google.pubsublite.kafka.sink;
 
-import com.google.cloud.pubsublite.CloudZone;
-import com.google.cloud.pubsublite.ProjectPath;
-import com.google.cloud.pubsublite.PublishMetadata;
-import com.google.cloud.pubsublite.TopicName;
-import com.google.cloud.pubsublite.TopicPath;
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.pubsublite.*;
+import com.google.cloud.pubsublite.cloudpubsub.PublisherSettings;
 import com.google.cloud.pubsublite.internal.Publisher;
-import com.google.cloud.pubsublite.internal.wire.PubsubContext;
 import com.google.cloud.pubsublite.internal.wire.PubsubContext.Framework;
 import com.google.cloud.pubsublite.internal.wire.RoutingPublisherBuilder;
 import com.google.cloud.pubsublite.internal.wire.SinglePartitionPublisherBuilder;
+import com.google.cloud.pubsublite.internal.wire.PartitionPublisherFactory;
 import java.util.Map;
+
 import org.apache.kafka.common.config.ConfigValue;
 
 class PublisherFactoryImpl implements PublisherFactory {
 
   private static final Framework FRAMEWORK = Framework.of("KAFKA_CONNECT");
 
+  private PartitionPublisherFactory getPartitionPublisherFactory(TopicPath topic) {
```

Comment:
Why do we need a custom PartitionPublisherFactory instead of using a default single partition publisher factory?

Reply:
Could you link to what you are suggesting?